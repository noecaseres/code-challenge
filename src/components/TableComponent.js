import React, { useEffect } from 'react';
import queryString from 'query-string';
import {useNavigate, useLocation} from 'react-router-dom';
import {Table, Form} from 'react-bootstrap';
import { useForm } from '../hooks/useForm';
import { getFile } from '../helpers/getFile';
import { TableList } from './TableList';


export const TableComponent = () => {

  const navigate = useNavigate();
  const location = useLocation();


  const {q = ''} = queryString.parse(location.search); //Parseo "q"
  const files = getFile(q);


  const { searchText, onInputChange } = useForm({
    searchText:q
  });


  const onSearchSubmit = () => {
    navigate(`?q=${searchText}`) //envio el queryparam
  }

  useEffect(() => {
    onSearchSubmit()
  }, [searchText])
  


  return (
    <>
      <Form onSubmit={ onSearchSubmit }>
          <input 
            type="text"
            placeholder='e.g: file1'
            className="w-100"
            name="searchText"
            autoComplete="off"
            value={ searchText }
            onChange={ onInputChange }
          />
      </Form>
      <Table striped bordered hover>
      <thead>
        <tr>
          <th>File Name</th>
          <th>Text</th>
          <th>Number</th>
          <th>Hex</th>
        </tr>
      </thead>
      <tbody>
        {
          files.map(file=>(
            <TableList
            key={file.id} 
            {...file}
            />
          ))
          }
      </tbody>
    </Table>
    </>
  )
}
