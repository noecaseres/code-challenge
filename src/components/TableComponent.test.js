
import { render, screen} from "@testing-library/react"
import { MemoryRouter } from "react-router-dom";
import { TableComponent } from './TableComponent';

 describe('Pruebas en <TableComponent/>', () => { 
 
     test('debe mostrarse correctamente con valores por default',()=>{
         const {container} =render(
            <MemoryRouter>
                <TableComponent/>
            </MemoryRouter>
         );
         expect (container).toMatchSnapshot();

     });

     test('debe mostrar file2 y el input con el valor del queryString',()=>{
        render(
           <MemoryRouter initialEntries={['/?q=file2']}>
               <TableComponent/>
           </MemoryRouter>
        );  
        const input = screen.getByRole('textbox');
        expect(input.value).toBe('file2')
    })
  })