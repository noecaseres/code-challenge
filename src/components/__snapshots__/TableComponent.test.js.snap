// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TableComponent/> debe de mostrarse correctamente con valores por default 1`] = `
<div>
  <form
    class=""
  >
    <input
      autocomplete="off"
      class="w-100"
      name="searchText"
      placeholder="e.g: file1"
      type="text"
      value=""
    />
  </form>
  <table
    class="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          File Name
        </th>
        <th>
          Text
        </th>
        <th>
          Number
        </th>
        <th>
          Hex
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          file1.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075909
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f765
        </td>
      </tr>
      <tr>
        <td>
          file1.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075910
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f765
        </td>
      </tr>
      <tr>
        <td>
          file1.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075911
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f765
        </td>
      </tr>
      <tr>
        <td>
          file2.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075912
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f764
        </td>
      </tr>
      <tr>
        <td>
          file2.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075913
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f764
        </td>
      </tr>
      <tr>
        <td>
          file3.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075914
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f763
        </td>
      </tr>
      <tr>
        <td>
          file3.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075915
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f763
        </td>
      </tr>
      <tr>
        <td>
          file3.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075916
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f763
        </td>
      </tr>
      <tr>
        <td>
          file3.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075917
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f763
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Pruebas en <TableComponent/> debe mostrarse correctamente con valores por default 1`] = `
<div>
  <form
    class=""
  >
    <input
      autocomplete="off"
      class="w-100"
      name="searchText"
      placeholder="e.g: file1"
      type="text"
      value=""
    />
  </form>
  <table
    class="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          File Name
        </th>
        <th>
          Text
        </th>
        <th>
          Number
        </th>
        <th>
          Hex
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          file1.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075909
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f765
        </td>
      </tr>
      <tr>
        <td>
          file1.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075910
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f765
        </td>
      </tr>
      <tr>
        <td>
          file1.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075911
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f765
        </td>
      </tr>
      <tr>
        <td>
          file2.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075912
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f764
        </td>
      </tr>
      <tr>
        <td>
          file2.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075913
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f764
        </td>
      </tr>
      <tr>
        <td>
          file3.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075914
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f763
        </td>
      </tr>
      <tr>
        <td>
          file3.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075915
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f763
        </td>
      </tr>
      <tr>
        <td>
          file3.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075916
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f763
        </td>
      </tr>
      <tr>
        <td>
          file3.csv
        </td>
        <td>
          RgTya
        </td>
        <td>
          64075917
        </td>
        <td>
          70ad29aacf0b690b0467fe2b2767f763
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
